gfx.cpp

```diff
@@ -965,11 +965,29 @@ class GfxInternal
         window_width_ = window_rect.right - window_rect.left;
         window_height_ = window_rect.bottom - window_rect.top;
 
+        IDXGIOutput *output;
+        DXGI_FORMAT  output_format = DXGI_FORMAT_R8G8B8A8_UNORM;
+        DXGI_COLOR_SPACE_TYPE output_type   = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+        if(adapter_->EnumOutputs(0, &output) != DXGI_ERROR_NOT_FOUND)
+        {
+            IDXGIOutput6 *output6 = nullptr;
+            output->QueryInterface(&output6);
+            DXGI_OUTPUT_DESC1 output_desc;
+            output6->GetDesc1(&output_desc);
+            if(output_desc.BitsPerColor > 8)
+                output_format = DXGI_FORMAT_R10G10B10A2_UNORM;
+            /*if(output_desc.ColorSpace == DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020)
+            {
+                output_type = DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+                output_format = DXGI_FORMAT_R10G10B10A2_UNORM;
+            }*/
```

Comment:
These are the relevant lines for enabling a HDR display output. This should be wrapped in a user option as we dont want to auto set HDR as it requires user side changes. I didnt add such as user option as I have no way to test it and didnt know how you prefered to do it. The user option would be optional (i.e. enable HDR if available) which would require an additional function/return that can be used to check if HDR was in use. This is complicated further by windows unfortunate handling of multiple displays as it possible for a window to be dragged between displays were only 1 is HDR requiring additional code to check for this and then change the swapchain accordingly and notify the user. This still falls down though when a window is straddling 2 monitors where only 1 is HDR. So additional code is required if HDR is to be supported in windowed mode or when windows are draggable